python/test-data/testTypesCollections4.err

```diff
@@ -0,0 +1,20 @@
+untypy.error.UntypyTypeError
+given:    42
+expected: value of type str
+
+context: foo(l: list[Callable[[], str]]) -> list[str]
+                                  ^^^
+declared at: test-data/testTypesCollections4.py:4
```

Comment:
This is the location where the "Function" was declared. In case of a none-lambda function there might be type annotations declared there.
I didn't want to handle this case separately.

Reply:
I see that there are situations in which multiple declarations contribute to an error (for example, if a class incorrectly implements a protocol). But in this situation, the signature of `foo(l: list[Callable[[], str]]) -> list[str]` is violated and there is exactly one location where this signature is defined. Even if the second location defines a non-lambda function with a type annotation, the signature being violated is still defined at only one location.

(Don't get me wrong: this "problem" is of minor importance. I merely want to understand the rationale behind.)